pandas/core/indexes/base.py

```diff
@@ -3158,9 +3158,9 @@ def _convert_slice_indexer(self, key: slice, kind=None):
 
         # validate iloc
         if kind == "iloc":
-            self._validate_indexer("slice", key.start, "iloc")
-            self._validate_indexer("slice", key.stop, "iloc")
-            self._validate_indexer("slice", key.step, "iloc")
+            self._validate_indexer("positional", key.start, "iloc")
+            self._validate_indexer("positional", key.stop, "iloc")
+            self._validate_indexer("positional", key.step, "iloc")
```

Comment:
Since this is for iloc, changing to positional (otherwise you can get confusing error message as "cannot do slice indexing on FloatIndex with indexer of type float" (which for label-based slice indexing would of course be totally fine)